webapp/channels/src/tests/react_testing_utils.tsx

```diff
@@ -25,3 +37,66 @@ export const renderWithIntlAndStore = (component: React.ReactNode | React.ReactN
         </IntlProvider>,
     );
 };
+
+export const renderWithFullContext = (component: React.ReactNode | React.ReactNodeArray, initialState: DeepPartial<GlobalState> = {}, locale = 'en') => {
+    // We use a redux-mock-store store for testing, but we set up a real store to ensure the initial state is complete
+    const testState = configureStore(initialState).getState();
+
+    // Store these in an object so that they can be maintained through rerenders
```

Comment:
I made sure to write unit tests to confirm that this actually worked how I wanted. I'm pretty sure that, if I didn't use an object to wrap these, the arrow functions used for `rerender` and such would've captured their original values and made it impossible to update them.

In retrospect, I feel like I could've a class for this and saved these as member variables, but I'm not going to bother changing that now